### PR TITLE
[feature] Allow dynamic segment in api redirect action 

### PR DESCRIPTION
**Pull request by donrestarone**
# Allow dynamic segment in api redirect action

This feature allows Ruby evaluation in the redirect action so you can do conditional redirects and other fanciness: 

Screen Shot 2022-07-08 at 11 14 21 AM

---
## How it works

The implicit return value of the evaluated statement should be a valid path in the application. As illustrated above, you can use an absolute path (like: `'/'`, or a Rails route helper like `dashboard_path`)

---

Addresses: https://github.com/restarone/violet_rails/issues/873

## Video demo

https://user-images.githubusercontent.com/35935196/178011977-82aab515-87b5-49a3-8591-f8ba3535d9eb.mp4

## The evaluated output of the dynamic_url segment  is saved in the lifecycle message
image

### CODE REVIEW

**Comment by donrestarone** [comment]
@alis-khadka this is on its way to Heroku for UAT

**Comment by donrestarone** [comment]
@alis-khadka here's the UAT feedback: 

## 1. Add inline-documentation for the dynamic segment
When the user selects dynamic_url they should be shown the variables that can be used in this segment (eg: `api_resource`, `rails route helpers eg: root_path`)
Screen Shot 2022-07-08 at 11 00 47 AM


## 2. the lifecycle message should include the evaluated output of the dynamic_url segment:

Screen Shot 2022-07-08 at 10 59 33 AM

File: app/views/comfy/admin/api_actions/_form.html.haml [hunk 1]
```diff
@@ -19,11 +19,20 @@
...
+              \\#{api_resource.properties['your_custom_attribute_here']}
+        .mt-2.redirect-field-guide{ style: "#{'display: none' unless (resource.redirect? && resource.dynamic_url?)}"}
+          %small.d-block
+            %strong
+              Available Variables:
+            api_resource, rails-route-helpers (example: root_path)
+          %small.d-block
+            Dynamic:
+            %code
+              \\#{ api_resource.properties['your_custom_attribute_here'] == 'true' ? rooth_path : some_other_path }
```
Comment (donrestarone) [inline app/views/comfy/admin/api_actions/_form.html.haml:35]: @alis-khadka  please replace `rooth_path` with `comfy_admin_cms_path`
Screenshot from 2022-07-10 13-52-58

File: test/controllers/resource_controller_test.rb [hunk 2]
```diff
@@ -192,6 +192,51 @@ class ResourceControllerTest < ActionDispatch::IntegrationTest
...
+      actions_count = @api_namespace.create_api_actions.size
+      assert_difference "@api_namespace.executed_api_actions.count", actions_count do
+        post api_namespace_resource_index_url(api_namespace_id: @api_namespace.id, params: payload)
+        assert_response :redirect
+        assert_redirected_to api_namespaces_path
+      end
+    end
+  end
+
+  test 'should allow #create and should not redirect by evaluating redirect_url if redirect_type is not dynamic_url' do
```
Comment (donrestarone) [inline test/controllers/resource_controller_test.rb:217]: @alis-khadka Could you explain this test case to me? I dont think I understand it

File: test/controllers/resource_controller_test.rb [hunk 3]
```diff
@@ -192,6 +192,51 @@ class ResourceControllerTest < ActionDispatch::IntegrationTest
     end
   end
```
Comment (donrestarone) [inline test/controllers/resource_controller_test.rb:193]: @alis-khadka could you add a test that verifies the-non dynamic case? Where we simply redirect to a CMS page 

See below for example: 
Screenshot from 2022-07-10 14-01-15

File: test/controllers/resource_controller_test.rb (hunk 2 above)
Comment (alis-khadka) [reply test/controllers/resource_controller_test.rb:217]: In this test, I am setting the redirect_url to: `"\#{ api_resource.properties['flag'] == 'true' ? dashboard_path : api_namespaces_path }"` but it is not a dynamic redirect-type. 
So, when the redirection is being done, it should redirect explicitly to `"\#{ api_resource.properties['flag'] == 'true' ? dashboard_path : api_namespaces_path }"` **without evaluating it**.

**Comment by donrestarone** [comment]
## UAT result on Heroku

1. General regression ✔️ 
2. dynamic redirect ✔️

**Pull request merged**